### ISSUE TITLE
Crash in CLI at particular point

### ISSUE THREAD

**Issue by cooldude77** [composer/composer#7307]
Hi,

 I just re- installed composer on my machine( after a disk issue)  and I am using XAMPP PHP 5.6.3 on Windows 10

My `composer.json`:

```json
{
  "name": "symfony/framework-standard-edition",
  "license": "MIT",
  "type": "project",
  "description": "The \"Symfony Standard Edition\" distribution",
  "autoload": {
    "psr-4": {
      "": "src/"
    },
    "files": [
      "app/AppKernel.php"
    ]
  },
  "require": {
    "php": ">=5.6.0",
    "symfony/symfony": "2.8.*",
    "doctrine/orm": "^2.4.8",
    "doctrine/doctrine-bundle": "~1.4",
    "symfony/swiftmailer-bundle": "~2.3",
    "symfony/monolog-bundle": "~2.4",
    "sensio/distribution-bundle": "~5.0",
    "sensio/framework-extra-bundle": "^3.0.2",
    "incenteev/composer-parameter-handler": "~2.0",
    "twig/extensions": "^1.3",
    "symfony/assetic-bundle": "^2.8",
    "sensio/generator-bundle": "^3.0",
    "friendsofsymfony/jsrouting-bundle": "^1.6",
    "symfony/var-dumper": "^3.0",
    "symfony/event-dispatcher": "^3.0",
    "vich/uploader-bundle": "^1.0",
    "jms/serializer-bundle": "^1.1",
    "friendsofsymfony/rest-bundle": "^1.7",
    "escapestudios/wsse-authentication-bundle": "dev-master",
    "willdurand/hateoas-bundle": "^1.1",
    "knplabs/knp-paginator-bundle": "^2.5",
    "deployer/deployer": "dev-master",
    "isometriks/spam-bundle": "dev-master",
    "sonata-project/seo-bundle": "^2.1",
    "endroid/gcm": "1.3.*",
    "csa/guzzle-bundle": "2.0",
    "guzzlehttp/guzzle": "^6.3"
  },
  "require-dev": {
    "symfony/phpunit-bridge": "^3.3",
    "phpunit/phpunit": "^5.2",
    "nelmio/alice": "^2.1",
    "liip/functional-test-bundle": "^1.5",
    "doctrine/doctrine-fixtures-bundle": "^2.3",
    "hautelook/alice-bundle": "^1.2",
    "mockery/mockery": "^0.9.9"
  },
  "repositories": [
    {
      "type": "vcs",
      "url": "https://github.com/Instanect/EscapeWSSEAuthenticationBundle"
    },
    {
      "type": "vcs",
      "url": "https://github.com/cooldude77/deployer"
    }
  ],
  "scripts": {
    "post-install-cmd": [
      "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
    ],
    "post-update-cmd": [
      "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
    ]
  },
  "config": {
    "bin-dir": "bin"
  },
  "extra": {
    "symfony-app-dir": "app",
    "symfony-web-dir": "web",
    "symfony-assets-install": "relative",
    "incenteev-parameters": {
      "file": "app/config/parameters.yml"
    },
    "branch-alias": {
      "dev-master": "2.8-dev"
    }
  }
}

```
Output of `composer diagnose`:

```
Checking composer.json: WARNING
Defining autoload.psr-4 with an empty namespace prefix is a bad idea for performance
require.escapestudios/wsse-authentication-bundle : unbound version constraints (dev-master) should be avoided
require.deployer/deployer : unbound version constraints (dev-master) should be avoided
require.isometriks/spam-bundle : unbound version constraints (dev-master) should be avoided
require.csa/guzzle-bundle : exact version constraints (2.0) should be avoided if the package follows semantic versioning
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.6.4
PHP version: 5.6.3
PHP binary path: C:\xampp\php\php.exe
```

When I run this command:

`composer update -vvv
`
I get the following output:

```
The composer update command crash at following location exactly 

Reading C:/Users/AKS/AppData/Local/Composer/repo/https---packagist.org/provider-jms$di-extra-bundle.json from cache
```
. I have 16 GB ram on my local machine.
I have put `memory_limit=-1`  and ran it again but same issue persist and PHP error log says
`
[04-May-2018 12:55:12 Asia/Kolkata] PHP Fatal error:  Out of memory (allocated 1396965376) (tried to allocate 268435456 bytes) in phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Solver.php on line 220`

**Comment by alcohol** [comment]
You need to configure PHP to allow for the consumption of more memory (either in your php.ini or by passing `-d memory_limit=-1` on the command line -- though I am not sure how one can do that on Windows when installing Composer through our provided installer).

**Comment by cooldude77** [comment]
@alcohol I have already done that in php.ini as a last measure and it did not work ( as mentioned in the issue)

**Comment by alcohol** [comment]
Yes but it does not look like you modified the right `php.ini` because the error you reported still shows a very low value.

**Comment by cooldude77** [comment]
Yeah, good catch.. Let me check other php,ini as well . Thanks :)

**Comment by xabbuh** [comment]
@cooldude77 you can use the `--ini` option of the PHP binary to find out which file is actually loaded

**Comment by cooldude77** [comment]
Many of us use XAMPP in order to develop projects on Windows. I was using the same. Unfortunately it is only available for 32 bit systems . I read in another issue that 32 bit allows a max of 2 GB to be used irrespective of memory_limit setting. ( and the update uses above 2GB)
So I tried updating it for 64 bit and I was unsuccessful. 

I installed WAMP next on my machine and after a few tries , restarted it . And voila , it worked like a charm ( after setting memory , of course). 

So this is what a person who is using composer on windows should try to do . 
Thanks all for your help and suggestion. Closing the issue.
